### PR TITLE
Check for mpfr.h in configure.ac

### PR DESCRIPTION
**Pull request by scemama**

### CODE REVIEW

**Review by titeup — APPROVED** [review]

File: configure.ac [hunk 1]
```diff
@@ -95,6 +95,10 @@ fi
 AC_CHECK_LIB([c], [exit], , AC_MSG_ERROR([Could not find c library]))
 AC_CHECK_LIB([m], [sin], , AC_MSG_ERROR([Could not find math library]))
 
+AC_CHECK_HEADERS([mpfr.h], [],
+                    [AC_MSG_ERROR("mpfr.h not found. Please install the libmpfr-dev deb package.")])
```
[the remark this answers is not in the document]
Comment (scemama) [reply configure.ac]: Done!

**Comment by scemama** [comment]
2 reasons:

1. In the install instructions, you have it in the list of apt-get packages.
2. I had the following error:
```
$ make
make  all-recursive
make[1]: Entering directory '/home/scemama/TREX/verificarlo'
Making all in src/
make[2]: Entering directory '/home/scemama/TREX/verificarlo/src'
Making all in common
make[3]: Entering directory '/home/scemama/TREX/verificarlo/src/common'
Making all in rng
make[4]: Entering directory '/home/scemama/TREX/verificarlo/src/common/rng'
make[4]: Nothing to be done for 'all'.
make[4]: Leaving directory '/home/scemama/TREX/verificarlo/src/common/rng'
make[4]: Entering directory '/home/scemama/TREX/verificarlo/src/common'
  AR       libvfc_probes.a
ar: `u' modifier ignored since `D' is the default (see `U')
  AR       libvfc_probes_f.a
ar: `u' modifier ignored since `D' is the default (see `U')
  CCLD     libvprec_tools.la
  CCLD     libvfc_hashmap.la
make[4]: Leaving directory '/home/scemama/TREX/verificarlo/src/common'
make[3]: Leaving directory '/home/scemama/TREX/verificarlo/src/common'
Making all in libvfcfuncinstrument
make[3]: Entering directory '/home/scemama/TREX/verificarlo/src/libvfcfuncinstrument'
  CXX      libvfcfuncinstrument_la-libVFCFuncInstrument.lo
  CXXLD    libvfcfuncinstrument.la
make[3]: Leaving directory '/home/scemama/TREX/verificarlo/src/libvfcfuncinstrument'
Making all in libvfcinstrument
make[3]: Entering directory '/home/scemama/TREX/verificarlo/src/libvfcinstrument'
  CXX      libvfcinstrument_la-libVFCInstrument.lo
In file included from /usr/lib/llvm-7/include/llvm/IR/IRBuilder.h:19,
                 from libVFCInstrument.cpp:23:
/usr/lib/llvm-7/include/llvm/ADT/ArrayRef.h: In instantiation of ‘llvm::ArrayRef<T>::ArrayRef(const std::initializer_list<_Tp>&) [with T = llvm::Value*]’:
libVFCInstrument.cpp:430:57:   required from here
/usr/lib/llvm-7/include/llvm/ADT/ArrayRef.h:102:37: warning: initializing ‘llvm::ArrayRef<llvm::Value*>::Data’ from ‘std::initializer_list<llvm::Value*>::begin’ does not extend the lifetime of the underlying array [-Winit-list-lifetime]
  102 |     : Data(Vec.begin() == Vec.end() ? (T*)nullptr : Vec.begin()),
  CXXLD    libvfcinstrument.la
make[3]: Leaving directory '/home/scemama/TREX/verificarlo/src/libvfcinstrument'
Making all in backends
make[3]: Entering directory '/home/scemama/TREX/verificarlo/src/backends'
Making all in interflop-ieee
make[4]: Entering directory '/home/scemama/TREX/verificarlo/src/backends/interflop-ieee'
  CC       libinterflop_ieee_la-interflop_ieee.lo
  CC       ../../common/libinterflop_ieee_la-printf_specifier.lo
  CC       ../../common/libinterflop_ieee_la-logger.lo
  CCLD     libinterflop_ieee.la
make[4]: Leaving directory '/home/scemama/TREX/verificarlo/src/backends/interflop-ieee'
Making all in interflop-mca
make[4]: Entering directory '/home/scemama/TREX/verificarlo/src/backends/interflop-mca'
  CC       libinterflop_mca_la-interflop_mca.lo
  CC       ../../common/libinterflop_mca_la-logger.lo
  CC       ../../common/libinterflop_mca_la-options.lo
  CCLD     libinterflop_mca.la
make[4]: Leaving directory '/home/scemama/TREX/verificarlo/src/backends/interflop-mca'
Making all in interflop-mca-int
make[4]: Entering directory '/home/scemama/TREX/verificarlo/src/backends/interflop-mca-int'
  CC       libinterflop_mca_int_la-interflop_mca_int.lo
  CC       ../../common/libinterflop_mca_int_la-logger.lo
  CC       ../../common/libinterflop_mca_int_la-options.lo
  CCLD     libinterflop_mca_int.la
make[4]: Leaving directory '/home/scemama/TREX/verificarlo/src/backends/interflop-mca-int'
Making all in interflop-cancellation
make[4]: Entering directory '/home/scemama/TREX/verificarlo/src/backends/interflop-cancellation'
  CC       libinterflop_cancellation_la-interflop_cancellation.lo
  CC       ../../common/libinterflop_cancellation_la-logger.lo
  CC       ../../common/libinterflop_cancellation_la-options.lo
  CCLD     libinterflop_cancellation.la
make[4]: Leaving directory '/home/scemama/TREX/verificarlo/src/backends/interflop-cancellation'
Making all in interflop-bitmask
make[4]: Entering directory '/home/scemama/TREX/verificarlo/src/backends/interflop-bitmask'
  CC       libinterflop_bitmask_la-interflop_bitmask.lo
interflop_bitmask.c:32:10: fatal error: mpfr.h: No such file or directory
   32 | #include <mpfr.h>
      |          ^~~~~~~~
compilation terminated.
make[4]: *** [Makefile:497: libinterflop_bitmask_la-interflop_bitmask.lo] Error 1
make[4]: Leaving directory '/home/scemama/TREX/verificarlo/src/backends/interflop-bitmask'
make[3]: *** [Makefile:379: all-recursive] Error 1
make[3]: Leaving directory '/home/scemama/TREX/verificarlo/src/backends'
make[2]: *** [Makefile:432: all-recursive] Error 1
make[2]: Leaving directory '/home/scemama/TREX/verificarlo/src'
make[1]: *** [Makefile:503: all-recursive] Error 1
make[1]: Leaving directory '/home/scemama/TREX/verificarlo'
make: *** [Makefile:398: all] Error 2

```

**Comment by pablooliveira** [comment]
Ok thanks @scemama ! I believe this include in bitmask could be removed, but in the meantime this PR should be merged. Thanks !

**Comment by titeup** [comment]
> mpfr dependency was checked before but the check was removed since we theoretically should not depend on it anymore (after removing the original libmca backend).
> 
> Since you are adding the check @scemama, it means it is needed again during the build. Could you please tell us where the build failed without it ? It would be useful to understand which piece of verificarlo is still asking for mpfr. Is it during tests ?
> 
> Thanks !

It appears in many test scripts (py) and bitmask.c according to github search.

**Comment by titeup** [comment]
> > mpfr dependency was checked before but the check was removed since we theoretically should not depend on it anymore (after removing the original libmca backend).
> > Since you are adding the check @scemama, it means it is needed again during the build. Could you please tell us where the build failed without it ? It would be useful to understand which piece of verificarlo is still asking for mpfr. Is it during tests ?
> > Thanks !
> 
> It appears in many test scripts (py) and bitmask.c according to github search.

Note that after looking at the bitmask.c code quickly I am not sur we are really using it...

**Pull request merged**